server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -964,6 +964,15 @@ private UserVm rebootVirtualMachine(long userId, long vmId, boolean enterSetup)
         if (vm.getState() == State.Running && vm.getHostId() != null) {
             collectVmDiskStatistics(vm);
             collectVmNetworkStatistics(vm);
+
+            if (forced) {
+                Host vmOnHost = _hostDao.findById(vm.getHostId());
+                if (vmOnHost == null || vmOnHost.getResourceState() != ResourceState.Enabled || vmOnHost.getStatus() != Status.Up ) {
```

Comment:
@sureshanaparti is this a new behaviour, did it use to throw exception/fail when host is not enabled or up without forced=true?

Reply:
@rhtyd Start VM will fail when host is not Enabled and not Up. As the force reboot stops and starts the VM, the start VM would fail, and so it's safe to not allow when force reboot when host is not Enabled and not Up.